server/src/main/java/com/lunchtime/config/WebSecurityConfig.java

```diff
@@ -0,0 +1,70 @@
+package com.lunchtime.config;
+
+import com.lunchtime.security.JwtAuthenticationTokenFilter;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.NoOpPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    //TODO You can use such stuff for debug or local env, but it is not suitable for production
+    // if you are not using token
+    // authentication. I can suggest to move this setting to application.properties.
+    // Link for help:
+    // https://stackoverflow.com/questions/44824382/how-to-disable-csrf-in-spring-using-application-properties
+
+    @Autowired
+    private UserDetailsService myUserDetailsService;
+    @Autowired
+    private JwtAuthenticationTokenFilter jwtAuthenticationTokenFilter;
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(myUserDetailsService);
+    }
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return NoOpPasswordEncoder.getInstance();
+    }
+
+    @Override
+    @Bean
+    public AuthenticationManager authenticationManagerBean() throws Exception {
+        return super.authenticationManagerBean();
+    }
+
+    @Override
+    protected void configure(HttpSecurity httpSecurity) throws Exception {
+        httpSecurity.csrf().disable().cors()
+            .and().authorizeRequests()
+            .antMatchers("/api/persons").permitAll()
+            .and().authorizeRequests()
+            .antMatchers("/api/restaurants/**","/api/events/**").permitAll().and()
+            .authorizeRequests().antMatchers("/api/authenticate").permitAll()
+            .anyRequest().authenticated().and()
+            .exceptionHandling().and().sessionManagement()
+            .sessionCreationPolicy(SessionCreationPolicy.STATELESS);
+        httpSecurity.addFilterBefore(jwtAuthenticationTokenFilter, UsernamePasswordAuthenticationFilter.class);
```

Comment:
We should reconfigure settings here. For example, as an authorized user I am allowed to get new generated token while I am logged in already. I mean I have a permission to enter `/login` page on client side while already authenticated and when enter correct email and password on ligin form I get another jwt-token.

Also redundant lines here

Reply:
to generate a refresh token this is not a valid thought i think
clean lines are here only for code readability and no extra